### PR TITLE
Make tokens upgradeable

### PR DESCRIPTION
**Pull request by sullof**
This makes SideToken upgradeable.

### CODE REVIEW

File: contracts/mocks/WeedTokenMock.sol [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "../token/SideToken.sol";
+
+contract WeedTokenMock is SideToken, UUPSUpgradeable {
```
Comment (superpower007) [inline contracts/mocks/WeedTokenMock.sol:7]: can WeedTokenMock be derived from WeedToken

**Comment by superpower007** [comment]
any reasons these tokens need to be upgradable other than bug fixes?

File: contracts/mocks/WeedTokenMock.sol (hunk 1 above)
Comment (sullof) [reply contracts/mocks/WeedTokenMock.sol:7]: Not without adding more unnecessary code to the WeedToken

**Comment by sullof** [comment]
BTW, that mock is necessary only during development. It is not exactly a mock, it is a different version of the weed token for dev purposes.

**Pull request merged**